platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/artifact/ArtifactSet.java

```diff
@@ -27,5 +27,12 @@ public interface ArtifactSet {
     /**
      * Selects the artifacts of this set that meet the given criteria. Implementation should be eager where possible, so that selection happens immediately, but may be lazy.
      */
-    ResolvedArtifactSet select(ArtifactVariantSelector variantSelector, ArtifactSelectionSpec spec);
+    default ResolvedArtifactSet select(ArtifactVariantSelector variantSelector, ArtifactSelectionSpec spec) {
```

Comment:
This overload can be removed, I just left it for now to make less tests need updating